measures/src/Data/Measure/ByteSize.hs

```diff
@@ -0,0 +1,328 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+
+-- | A measure for practical byte sizes.
+--
+-- 'ByteSize' is for summation, which might overflow. The other types are for
+-- storing, serializing, comparing, etc the results of calculations that did
+-- not overflow.
+--
+-- Import this module qualified.
+--
+-- > import           Data.Measure.ByteSize (ByteSize)
+-- > import qualified Data.Measure.ByteSize as ByteSize
+module Data.Measure.ByteSize (
+    ByteSize,
+    -- * Observers
+    compare,
+    isOverflowed,
+    -- * Safe result types
+    ByteSize8 (ByteSize8, unByteSize8),
+    ByteSize16 (ByteSize16, unByteSize16),
+    ByteSize32 (ByteSize32, unByteSize32),
+    ByteSize64 (ByteSize64, unByteSize64),
+    -- * Conversions
+    ByteSizeFrom,
+    ByteSizePartialFrom,
+    ByteSizeTo,
+    from,
+    partialFrom,
+    partialFromDefault,
+    to,
+    -- * Unstable and unsafe
+    unsafeCoercionWord64,
+  ) where
+
+import           Cardano.Binary (FromCBOR, ToCBOR)
+import           Control.DeepSeq (NFData)
+import           Data.Coerce (coerce)
+import           Data.DerivingVia (InstantiatedAt (InstantiatedAt))
+import           Data.Int (Int8, Int16, Int32, Int64)
+import           Data.Maybe.Strict (StrictMaybe (..))
+import           Data.Measure (BoundedMeasure, Measure)
+import qualified Data.Measure as Measure
+import           Data.Type.Coercion (Coercion (Coercion))
+import           Data.Word (Word8, Word16, Word32, Word64)
+import           GHC.Generics (Generic)
+#if __GLASGOW_HASKELL__ < 900
+-- Use the GHC version here because this is compiler dependent, and only indirectly lib dependent.
+import           GHC.Natural (Natural)
+#endif
```

Comment:
Why CPP here?
```suggestion
import           Numeric.Natural (Natural)
```

Reply:
I don't know; I merely copied it from an existing module in the package.